packager/packager.cc

```diff
@@ -279,23 +279,139 @@ class PackageInfoImpl final : public core::packages::PackageInfo {
     explicit PackageInfoImpl(const PackageInfoImpl &) = default;
     PackageInfoImpl &operator=(const PackageInfoImpl &) = delete;
 
-    optional<core::AutocorrectSuggestion> addImport(const core::GlobalState &gs, const PackageInfo &pkg,
-                                                    bool isTestImport) const {
-        auto &info = PackageInfoImpl::from(pkg);
-        auto it = absl::c_find_if(importedPackageNames, [&info](auto &import) { return import.name == info.name; });
-        if (it != importedPackageNames.end()) {
-            if (!isTestImport && it->type == core::packages::ImportType::Test) {
-                return convertTestImport(gs, info, core::Loc(fullLoc().file(), it->name.loc));
+    // What order should these packages be in the import list?
+    // Returns -1 if a should come before b, 0 if they are equivalent, and 1 if a should come after b.
+    //
+    // This method replicates the logic used at Stripe to order packages and thus has all of the associated "quirks".
+    // In particular, the ordering in a given package is a function of the strictness level of the package, and it is
+    // not a simple "false < layered < layered_dag < dag" ordering. The ordering is as follows:
+    // For strictDependenciesLevel::False:
+    // - layering violations
+    // - imports to 'false' packages
+    // - imports to 'layered' or stricter packages
+    // For strictDependenciesLevel::Layered and LayeredDag:
+    // - layering violations
+    // - imports to 'false' packages
+    // - imports to 'layered' or 'layered_dag' packages
+    // - imports to 'dag' packages
+    // For strictDependenciesLevel::Dag:
+    // - layering violations
+    // - imports to 'false', 'layered', or 'layered_dag' packages
+    // - imports to 'dag' packages
+    // TODO(neil): explain the rationale behind this ordering
+    // Within each group by strictness, packages are sorted by alphabetical order.
+    // TODO(neil): implement alphabetical sort.
```

Comment:
It seems weird to have a comment about what the function does and then have a TODO to implement that.  Just have the TODO, and when alphabetical sort happens, we can change the TODO into the actual description?